### DISCUSSION TITLE
Update an entry when another entry links to it

### DISCUSSION THREAD

**Discussion by ARCHES-Academy** [statamic/cms#4464] [Q&A] [answered]
Consider this scenario:

An entry (A) blueprint contains an Entry link field, and a user uses this field to link an entry (A) to another entry (B).

Is there an option to automatically update the linked entry (B) with a reference to the entry (A) being linked from?

**Comment by edalzell** [comment, marked as answer]
You could write a listener for `EntrySaved` and handle it in there?

**Reply by ARCHES-Academy** [reply]
Is the entry given when that event fires an instance of `Statamic\Facades\Entry`?

``` php
public function handle(EntrySaved $event)
{
    $event->entry;
}
```

Meaning, do I have access to all of the methods of that Facade, like `$event->entry->id()`,
or `$event->entry->value('content')`?

**Reply by duncanmcclean** [reply]
> Is the entry given when that event fires an instance of `Statamic\Facades\Entry`?
> 
> 
> 
> ``` php
> 
> public function handle(EntrySaved $event)
> 
> {
> 
>     $event->entry;
> 
> }
> 
> ```
> 
> 
> 
> Meaning, do I have access to all of the methods of that Facade, like `$event->entry->id()`,
> 
> or `$event->entry->value('content')`?

Yes, you'll have access to all the same methods.

**Reply by ARCHES-Academy** [reply]
Awesome, thanks for the guidance. For those that may be trying the same thing, here's the function that I wrote that is working for me.

``` php
public function handle(EntrySaved $event)
    {
        $entry = $event->entry;
        if ($entry->value('assignments') !== null) {
            $assignments = $entry->value('assignments');
            if (is_array($assignments)) {
                foreach ($assignments as $assignment) {
                    $a = Entry::find($assignment);
                    if ($a !== null) {
                        $a->set('lesson_plan', $entry->id());
                        $a->saveQuietly();
                    }
                }
            } else {
                $a = Entry::find($assignments);
                if ($a !== null) {
                    $a->set('lesson_plan', $entry->id());
                    $a->saveQuietly();
                }
            }
        }
    }
```

**Reply by edalzell** [reply]
FYI, your code can be tided up:

```php
if (! $assignments = $entry->value('assignments')) {
    return;
}

foreach (Arr:wrap($assignments) as $assignment) {
    if ($a = Entry::find($assignment)) {
        $a->set('lesson_plan', $entry->id())->saveQuietly();
	}
}
```

**Reply by ARCHES-Academy** [reply]
Nice, thanks for that. I'm still a bit of a beginner when it comes to PHP.
